app/controllers/student_teams_controller.rb

```diff
@@ -51,8 +51,7 @@ def view
         break
       end
     end
-
-    @teammate_review_allowed = true if @student.assignment.find_current_stage=='Finished' || @current_due_date&&(@current_due_date.teammate_review_allowed_id ==3 ||@current_due_date.teammate_review_allowed_id ==2) #late(2) or yes(3)
+    @teammate_review_allowed = true if @current_due_date&&(@current_due_date.teammate_review_allowed_id ==3 ||@current_due_date.teammate_review_allowed_id ==2) #late(2) or yes(3)
```

Comment:
Please take a look on "find_current_stage" in assignment.rb. If all the deadlines have passes, this method will return a string "Finished". We made the system allow teammate review even after the last round of deadline.